test/insert_test.dart

```diff
@@ -98,6 +98,43 @@ void main() {
 '''));
       expectYamlBuilderValue(doc, [0, 1, 2]);
     });
+
+    test('block insert', (){
+      final yamlEditor = YamlEditor('''
+# comment
+a:
+  - z:
+      x: 1
+      y: 2
+  - z:
+      x: 3
+      y: 4
+b:
+  - w:
+      m: 2
+      n: 4
+''');
+      yamlEditor.insertIntoList(['a'], 2, {'z' : {'x': 5, 'y': 6}});
+
+      expect(yamlEditor.toString(), equals('''
+# comment
+a:
+  - z:
+      x: 1
+      y: 2
+  - z:
+      x: 3
+      y: 4
+  - z:
+      x: 5
+      y: 6
+b:
+  - w:
+      m: 2
+      n: 4
+'''));
+    });
+
```

Comment:
May I propose an additional test along the lines of:

```dart
for (var i = 0; i < 3; i++) {
  test('block insert at $i', () {
    final yamlEditor = YamlEditor('''
a:
  - z:
      x: 1
      y: 2
  - z:
      x: 3
      y: 4
b: "next line"
''');
    // Check that this doesn't throw an internal error.
    yamlEditor.insertIntoList(['a'], i, {'z' : {'x': 5, 'y': 6}});
  });
}
```

If you look at the original issues you'll see that `insertIntoList` can throw an exception, if modification results in invalid YAML.
Specifically, it'll throw:
> Assertion failed: (package:yaml_edit) Modification did not result in expected result.

So while it's wonderful to have some tests where we check that the output is as expected, it's also really nice to have a lot of tests that tries different input without making assertions on the output. Such tests are easy to write a lot of, and they check that the `Assertion failed` error isn't thrown. So while it's not a perfect test, it's a test that won't break if we change the output of formatting a little -- that kind of test is easy to maintain.

Reply:
Added few more test cases considering different patterns.